### PR TITLE
Initialize project with components

### PR DESCRIPTION
**Pull request by krisbobo**
## Milestone: Initialize bookstore
*The following changes are being introduced:*
- Initialised React app.
- React routers with navigations
- Books view displays the list of books with the remove button(no functionality yet).
- Created a component called Book for displaying a single book.
- Minimal styling.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+- Node js installed on your computer
+- Npm Node package manager
+- Terminal
+
+### Setup
+
+Clone this repository to your desired folder:
+
+> https://github.com/krisbobo/maths-magicians.git
+
```
Comment (Gambit142) [inline README.md]: - [x] Great work setting up your _**README.md**_ file. The link that you provided for people to clone this repository is that of the **_Math Magician_** project. Kindly include the correct link. 👍

File: package.json [hunk 2]
```diff
@@ -9,6 +9,7 @@
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
     "react-scripts": "5.0.1",
+    "uuid": "^9.0.0",
     "web-vitals": "^2.1.4"
```
Comment (Gambit142) [inline package.json]: - [x] For you to use the `Routes`, `Route`, Link and other react-router-dom components, you need to install `react-router-dom`. You can install it by typing this in your terminal `npm i react-router-dom`. 👍

File: src/components/addBook.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import style from './AddBook.module.css';
+
```
Comment (Gambit142) [inline src/components/addBook.js]: - [x] The path to the CSS module is `'/addBook.module.css` and not `'./AddBook.module.css'`. Please pay attention to the capitalized `A` that you used and replace it with a small `a`. 👍 💻

File: src/components/displaybooks.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+/* eslint-disable */
+import React from 'react';
+import style from './displaybooks.module.css';
+
```
Comment (Gambit142) [inline src/components/displaybooks.js]: - [x] The path to the CSS module is `'/displaybook.module.css` and not `'./displaybooks.module.css'`. Please pay attention to the `s` at the end of displaybook and remove it. 👍💻

File: src/components/displaybooks.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+/* eslint-disable */
```
Comment (Gambit142) [inline src/components/displaybooks.js]: - [x] Kindly remove the `eslint` disabling comment at the beginning of this file. It is not good JavaScript practice to use eslint disabling comments. `Eslint` helps developers write clean and industry-standard JavaScript code. 👍

File: src/components/bookList.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
+import React, { useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import DisplayBooks from './displaybooks';
+
+export default function BooksList() {
+  // eslint-disable-next-line no-unused-vars
+  const [books, setBooks] = useState([
```
Comment (Gambit142) [inline src/components/bookList.js]: - [x] Kindly remove the `eslint` disabling comment at the beginning of this file. It is not good JavaScript practice to use eslint disabling comments. `Eslint` helps developers write clean and industry-standard JavaScript code. 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @krisbobo 👋, 

Good job so far!

## Highlights
- _**README.md**_ file is professional ✔️ 
- Pull request title and summary is descriptive ✔️
- Commit message are in imperative tense and capitalized ✔️ 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please, remember to tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: README.md (hunk 1 above)
Comment (krisbobo) [reply README.md]: Noted👍

File: package.json (hunk 2 above)
Comment (krisbobo) [reply package.json]: 👍

File: src/components/addBook.js (hunk 3 above)
Comment (krisbobo) [reply src/components/addBook.js]: 😕

File: src/components/bookList.js (hunk 6 above)
Comment (zieeco) [reply src/components/bookList.js]: - Hey, please, as suggested by the previous reviewer, kindly remove the `eslint` disable comment here as he's stated the reason above. Please, kindly work on this issue :pray:

File: src/components/displaybooks.js (hunk 5 above)
Comment (zieeco) [reply src/components/displaybooks.js]: - Hey, please, as suggested by the previous reviewer, kindly remove the `eslint` disable comment here as he's stated the reason above. Please, kindly work on this issue :pray:

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @krisbobo ,


Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the changes, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice setup. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._





## Optional suggestions
- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by omarsalem7** [comment]
Hi @krisbobo 😊, 
# STATUS: APPROVED! 🥇 💯 

giphy

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Descriptive pull request 
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**